GDJS/Runtime/pixi-renderers/layer-pixi-renderer.js

```diff
@@ -54,34 +62,44 @@ gdjs.LayerPixiRenderer.prototype.updatePosition = function() {
   this._pixiContainer.position.y += this._layer.getHeight() / 2;
 };
 
-gdjs.LayerPixiRenderer.prototype.updateVisibility = function(visible) {
+gdjs.LayerPixiRenderer.prototype.updateVisibility = function (visible) {
   this._pixiContainer.visible = !!visible;
 };
 
-gdjs.LayerPixiRenderer.prototype.updateTime = function() {
-  for(var filterName in this._filters) {
+gdjs.LayerPixiRenderer.prototype.updateTime = function () {
+  if (this._renderTexture) {
+    this.updateRenderTexture();
+  }
+
+  for (var filterName in this._filters) {
     var filter = this._filters[filterName];
     filter.update(filter.pixiFilter, this._layer);
   }
+
+  // Add Layer to lighting after adding effects to the layer,
+  // so that we can fix the blend mode.
+  if(!this._lightingSprite && this._isLightingLayer) {
+    this.replaceContainerWithSprite();
```

Comment:
Could we avoid doing this check at every update?

Reply:
Yes, it needs to be called after the filters are applied (because we need to change the blend mode of filters) and after all of the layers are added to the scene. So, maybe in the constructor,
https://github.com/4ian/GDevelop/blob/693b64cddfa8c3590a36a857e6f8a544d5fe653e/GDJS/Runtime/pixi-renderers/layer-pixi-renderer.js#L22-L24
after `this._setupFilters()`?

Reply:
Works in the constructor after `this._setupFilters()` :)

Reply:
I filters are added/removed (which can be the case with hot reloading), it might need to be updated... let's add a TODO in the code: `// TODO: Handle update of filters`.